core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -648,20 +775,50 @@ private[spark] class ExecutorAllocationManager(
      *
      * Note: This is not thread-safe without the caller owning the `allocationManager` lock.
      */
-    def pendingTasks(): Int = {
-      stageAttemptToNumTasks.map { case (stageAttempt, numTasks) =>
-        numTasks - stageAttemptToTaskIndices.get(stageAttempt).map(_.size).getOrElse(0)
+    def pendingTasksPerResourceProfile(rpId: Int): Int = {
```

Comment:
Is it better to refactor code to use a common function from line 778 to line 814? The following stuff can be reused.

```
      val pending = attempts.map { attempt =>
        val numTotalTasks = stageAttemptToNumTasks.getOrElse(attempt, 0)
        val numRunning = stageAttemptToTaskIndices.get(attempt).map(_.size).getOrElse(0)
        numTotalTasks - numRunning
      }.sum
```

Reply:
yes good idea, I'll make a utility function